slack_connector/main.py

```diff
@@ -76,16 +78,27 @@ def get_messages(api_token,channel_id):
 
                 result = client.conversations_history(channel = c)
                 conversation_history = result["messages"]
-                df_conversation_history = pd.DataFrame(conversation_history)
+                
 
-                if df_conversation_history.empty == True:
+                if conversation_history.empty == True:
                     raise NoMessageError()
 
+                df_conversation_history = pd.DataFrame(conversation_history)
                 df.append(df_conversation_history)
             
             except SlackApiError as e:
+
+                if e.response.data.get("error", "") == "not_in_channel":
+               
+                    logger.warning('Attempted to get messages for channel: {} that '
+                    'user is not in'.format(c))
+
+                    conversation_history = None
+            
+                else:
+                    raise e
             
-                logger.error("Error creating conversation list: {}".format(e))
```

Comment:
I suspect this logger.errror call should be called within the else: context before raising the e?  

As it is logging an error! 

```` python 
                else:
                    logger.error("Error creating conversation list: {}".format(e))
                    raise e
````

otherwise it will logged regardless outside the if, else context